### PR TITLE
feat: ignore config.*.local.y*ml

### PR DESCRIPTION
**Pull request by hanoii**
This is really a small addition, my goal here is so that an addon can configure something up and make sure it's going to be ignored.

As I am working on ddev/ddev-drupal-contrib I was thinking in providing a command to switch core. The readme says changing and env var on config.local.yaml, which obviously is fine, but If I wanted to automate this somewhow I was hoping to do something like:

`ddev init-drupal --version 11` which could create a `config.contrib.local.yaml`

With the following:

```yml
php: 8.3
corepack_enable: true
web_environment:
  - DRUPAL_CORE=^11
```

And that file will automatically be ignored.

This is very tangentially related to #6633

### CODE REVIEW

**Comment by hanoii** [comment]
I just thought that `config.contrib.local.yaml` will not be read, but it is!

**Comment by rfay** [comment]
`config.contrib_local.yaml` is your fastest (and most recommended) way out of this :)

**Comment by hanoii** [comment]
> `config.contrib_local.yaml` is your fastest (and most recommended) way out of this :)

but this is not current ignored, the PR is to be able to add something that I know will be ignored  by ddev, if it makes sense.

**Comment by rfay** [comment]
Ah, 
> make sure it's going to be ignored.

That may be something that should/could be figured out using the content of the files, rather than their names? (I didn't understand the intent first time around.)

**Comment by hanoii** [comment]
We use filenames a lot in general, I don't know, this looks easy enough?

**Review by rfay — CHANGES_REQUESTED** [review]
OK, I can live with this, but it needs docs in https://ddev.readthedocs.io/en/stable/users/extend/customization-extendibility/#extending-configyaml-with-custom-configyaml-files

File: docs/content/users/extend/customization-extendibility.md [hunk 1]
```diff
@@ -318,7 +318,7 @@ router_http_port: 8080
 
 Extra `config.*.yaml` files are loaded in lexicographic order, so `config.a.yaml` will be overridden by `config.b.yaml`.
 
-Team members may choose to use `config.local.yaml` for local non-committed config changes, for example. `config.local.yaml` is gitignored by default.
+Team members may choose to use `config.local.yaml` for local non-committed config changes, for example. `config.*.local.yaml` files can also be used to modify the order in which the local overrides are loaded. `config.local.yaml` and `config.*.local.yaml` are gitignored by default.
```
Comment (rfay) [inline docs/content/users/extend/customization-extendibility.md:321]: I guess I didn't understand this had anything to do with overrides? And how does it change order? Looks to me like it's just another config.*.yaml, which happens to be gitignored.

**Comment by hanoii** [comment]
@rfay docs added. I realized that this have the added benefit of one can locally alter addons configs by using something the lexicographic order. Unfortunately, `config.redis.yaml` comes after `config.redis.local.yaml`, but one can always do `config.redisoverrides.local.yaml` or whatever filename you want to make sure that comes after the one you want to override. I think any local file should come after the non-local ones, but that's another discussion.

File: docs/content/users/extend/customization-extendibility.md (hunk 1 above)
Comment (hanoii) [reply docs/content/users/extend/customization-extendibility.md:321]: @rfay see my comment bellow in the issue queue. I just realize that, that's why I mentioned it on the docs.

**Comment by rfay** [comment]
The way we do things so they sort right is `config.redis_extra.yaml`, for example. 

The way to *override* is to use [override_config](https://ddev.readthedocs.io/en/stable/users/configuration/config/#override_config)

**Comment by hanoii** [comment]
right, that's what I meant, or attempt to mean.  you can now do a `config.redis_extra.local.yaml` that will be gitignored by default. If you think what I wrote is confusing I can keep it simpler, will edit.

**Comment by hanoii** [comment]
@rfay this was actually what I did first.

**Review by rfay — APPROVED** [review]
Thanks, makes sense to me. I wish we hadn't done `config.local.y*ml` as opposed to `config.local.yaml` in the distant past whenever it was, but I guess that's where we are.

**Pull request merged**